### PR TITLE
elmPackages.elm-review: Init at 2.3.3

### PR DESCRIPTION
**Pull request by turboMaCk**
Elm-review is the most advanced linter for elm language. I think we might want to use it for nixos-search.

I had to temporarily generate node-env.nix file for elmPackages as we're waiting for nodePackages to get updated.

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/elm/packages/node-packages.json
Comment (turboMaCk) [inline pkgs/development/compilers/elm/packages/node-packages.json:2]: no follows .editorconfig rules

**Comment by domenkozar** [comment]
@GrahamcOfBorg build elmPackages

File: pkgs/development/compilers/elm/packages/node-composition.nix
Comment (SuperSandro2000) [inline pkgs/development/compilers/elm/packages/node-composition.nix:9]:
```suggestion
    inherit (pkgs) stdenv python2 util-linux runCommand writeTextFile;
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 105371` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>elmPackages.create-elm-app</li>
    <li>elmPackages.elm-analyse</li>
    <li>elmPackages.elm-language-server</li>
    <li>elmPackages.elm-review</li>
    <li>elmPackages.elm-upgrade</li>
  </ul>
</details>

File: pkgs/development/compilers/elm/packages/node-composition.nix
Comment (turboMaCk) [reply pkgs/development/compilers/elm/packages/node-composition.nix:9]: This file is generated by node2nix. I think this change comes from the new version. If we would revert this, we should also ditch `node-env.nix` I think.

Comment (turboMaCk) [reply pkgs/development/compilers/elm/packages/node-composition.nix:9]: @domenkozar what do you think about this? Should I edit generated files?

**Review by garbas — APPROVED** [review]
Looks good to me, but I'll let @domenkozar do the review and merge.

**Comment by turboMaCk** [comment]
I'll also open separate PR for marking aarch as bad platform for things that are failing there once this is merged.

File: pkgs/development/compilers/elm/packages/node-composition.nix
Comment (turboMaCk) [reply pkgs/development/compilers/elm/packages/node-composition.nix:9]: so i this PR I did not change any generated file. This suggestion is a change in file which was generated. This change in how node2nix generates file is also why we now need our own node-env.nix. I'm not sure if anything needs fixing because the newly generated files do work just fine. I think all we would need to avoid creating node-env.nix is would be to regenerate nodePackages as well Which I don't want to do to keep scope of this change small.

Furthermore I think it'll be better for node2nix maintainers if we would break the dependecy on their node-env.nix for a moment so they can do the transition to new format on their own terms without breaking elmPackages.

**Pull request merged**
